crates/service/src/status.rs

```diff
@@ -19,20 +25,84 @@ impl<S: Service> ServiceMonitor<S> {
     pub fn get_current(&self) -> S::Status {
         self.status_rx.borrow().clone()
     }
+
+    /// Converts this service monitor into a generic one.
+    pub fn into_generic(self) -> GenericStatusMonitor {
+        GenericStatusMonitor::new(self)
+    }
 }
 
 /// Service monitor type.
 ///
 /// This is intended to be object-safe so that we can have a collection of
 /// monitors for heterogeneous service types.
 pub trait StatusMonitor {
+    /// Fetches the latest status as a boxed `dyn Any`.
+    fn fetch_status_any(&mut self) -> anyhow::Result<AnyStatus>;
+
     /// Fetches the latest status as a JSON value.
-    fn fetch_status(&mut self) -> anyhow::Result<serde_json::Value>;
+    fn fetch_status_json(&mut self) -> anyhow::Result<serde_json::Value>;
```

Comment:
Intuitively these shouldn't need a `&mut self` since we are reading values.
What am I missing? Are we going to change the returned `status`?

Reply:
Oh good point.